### PR TITLE
Move vehicle flyability to a flag, allow modifying flying vehicles

### PR DESCRIPTION
**Pull request by anothersimulacrum**
#### Summary
SUMMARY: None

#### Purpose of change
Follow-up to #40467 
Instead of forbidding modification, use a tag on a vehicle to determine if it is airworthy. Doing any of the previously prevented activities will not set this tag to false.

#### Describe the solution
Add `vehicle::flyable`, `vehicle::is_flyable()` and `vehicle::set_flyable()` these either return or modify this value. This value defaults to true. 
Add `vehicle::would_prevent_flyable()` this determines if modifying a specific vehicle part would render this vehicle non-flyable.
In vehicle interaction, query if the player wants to render the vehicle unflyable, and if they consent to it, render the vehicle non-flyable.
In vehicle modification activities, bail out if the selected action would make the vehicle unflyable.
When determining if the player can change z levels, check if the vehicle is flyable, and if it is not, bail out.

#### Testing
Spawn in a helicopter. Attempt to uninstall a part that is not a curtain or seatbelt.
Spawn in a helicopter. Attempt to install a part that is not a curtain or seatbelt.
Spawn in a helicopter. Smash it up a bit, then attempt to repair a part that is not a curtain or seatbelt.
Say no to each of these actions. You can still fly the helicopter. Say yes to each of these actions. Be prevented from flying the helicopter.
Save and load, unflyable vehicles are still unflyable.
Attempt to deconstruct an untouched vehicle. It works fine.
Attempt to deconstruct an intact helicopter. It bails out.

#### Additional context
image
image
image

### CODE REVIEW

**Comment by mrkybe** [comment]
Seats should be repairable and removing doors would from a helicopter is a thing for some helicopters.  I don't see why mounting a storage trunk inside a helicopter would make it not airworthy either.

**Comment by Rogal-Dorn** [comment]
This seems like a much better solution and seems to have a clearer vision going forward, in my opinion at least. One problem I see is that the previously trivial to json out behavior now seems hardcoded. The ROTOR flag is the only place determining both flight and this airworthy check, I believe. Could a new flag be added or something similar that allows json edits to re enable the "gamey" helicopter modification?

File: src/activity_item_handling.cpp [hunk 1]
```diff
@@ -1124,10 +1124,14 @@ static activity_reason_info can_do_activity_there( const activity_id &act, playe
             for( vehicle_part *part_elem : parts ) {
                 const vpart_info &vpinfo = part_elem->info();
                 int vpindex = veh->index_of_part( part_elem, true );
-                // if part is not on this vehicle, or if its attached to another part that needs to be removed first.
+                // if part is not bnon this vehicle, or if its attached to another part that needs to be removed first.
```
Comment (ZhilkinSerg) [inline src/activity_item_handling.cpp:1127]:
```suggestion
                // if part is not on this vehicle, or if its attached to another part that needs to be removed first.
```

**Comment by Rogal-Dorn** [comment]
With the latest commit this seems to be much more logical. Thanks for putting in the work. 
In another PR it may be desirable to have other ways to lose airworthiness such as the hull taking too much damage. Should damaged rotors ever be able to fly? I don't know if there are any cases of that historically to refer to, but flying with a damaged rotor seems like a bad idea. Regardless, out of scope for this PR.

**Comment by nphyx** [comment]
This is a much better solution! I still think a lot of arguments could be made as to what should and shouldn't be repairable, but with this change those can be done incrementally without making flying vehicles useless to anyone without a pilot license in the meanwhile (at least they're good for parts again now).

**Pull request merged**